pandas/core/arrays/datetimes.py

```diff
@@ -713,12 +713,10 @@ def _sub_datetime_arraylike(self, other):
             assert is_datetime64_dtype(other)
             other = type(self)(other)
 
-        if not self._has_same_tz(other):
-            # require tz compat
-            raise TypeError(
-                f"{type(self).__name__} subtraction must have the same "
-                "timezones or no timezones"
-            )
+        try:
+            self._assert_tzawareness_compat(other)
+        except TypeError as e:
+            raise TypeError(str(e).replace("compare", "subtract"))
```

Comment:
instead of str.replace, can you edit the existing message that keeps `type(self).__name__`

Reply:
i did:
`raise type(error)(new_message) from error`

so we can support subclasses and get the stack trace. the alternative to this is to modify the 'args' of the existing error, but that seems like it would depend on the error being raised in the right way